py_src/ipyelk/nx/pipes.py

```diff
@@ -0,0 +1,193 @@
+from typing import Dict, Hashable, Iterator, Optional
+
+import ipywidgets as W
+import networkx as nx
+import traitlets as T
+
+from ipyelk.elements import Edge, HierarchicalElement, Label, Node, Port, Registry
+from ipyelk.elements.serialization import build_edge, build_shape_map, iter_elements
+
+from .. import diagram
+from ..elements import layout_options as opt
+from ..pipes import BrowserTextSizer, ElkJS, Pipe, Pipeline
+
+root_opts = opt.OptionsWidget(
+    options=[
+        opt.HierarchyHandling(),
+    ],
+).value
+label_opts = opt.OptionsWidget(
+    options=[opt.NodeLabelPlacement(horizontal="center", vertical="center")]
+).value
+node_opts = opt.OptionsWidget(
+    options=[
+        opt.NodeSizeConstraints(),
+    ],
+).value
+
+
+class NXSource(W.Widget):
+    graph: nx.MultiDiGraph = T.Instance(nx.MultiDiGraph)
+    hierarchy: nx.DiGraph = T.Instance(nx.DiGraph, allow_none=True)
+
+
+class NetworkxPipe(Pipe):
```

Comment:
Would it be more explicit to call this `Networkx2ElkJson(Pipe)`?  Or is that too much?